codecs/opus/src/main/java/org/restcomm/media/codec/opus/Decoder.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2011, Red Hat, Inc. and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.restcomm.media.codec.opus;
+
+import org.restcomm.media.spi.dsp.Codec;
+import org.restcomm.media.spi.format.Format;
+import org.restcomm.media.spi.format.FormatFactory;
+import org.restcomm.media.spi.memory.Frame;
+import org.restcomm.media.spi.memory.Memory;
+
+/**
+ * Implements Opus decoder.
+ */
+public class Decoder implements Codec {
+
+    private final static Format opus = FormatFactory.createAudioFormat("opus", 48000, 8, 1);
+    private final static Format linear = FormatFactory.createAudioFormat("linear", 8000, 16, 1);
+
+    private int j=0,i=0;
+    private int sourceLen=0,destinationLen=0;
+
+    /**
+     * (Non Java-doc)
+     * 
+     * @see org.mobicents.media.server.impl.jmf.dsp.Codec#getSupportedFormat().
+     */
+    public Format getSupportedInputFormat() {
+        return opus;
+    }
+
+    /**
+     * (Non Java-doc)
+     * 
+     * @see org.mobicents.media.server.impl.jmf.dsp.Codec#getSupportedFormat().
+     */
+    public Format getSupportedOutputFormat() {
+        return linear;
+    }
+
+    /**
```

Comment:
Erase non-javadoc comment (and that class does not exist in this branch).
Add `@Override` annotation.